### PR TITLE
Handling API Keys within the Plugin UI 

### PR DESCRIPTION
**Pull request by namithj**
# Pull Request

## What changed?

#36 
#33 
#26 
#13 

Renamed constant AP_HOSTS to AP_HOST which now accepts just one api host domain as a string `define( 'AP_HOST', 'api.aspirecloud.org' )`

Added a new constant AP_REMOVE_UI to disable configuring the plugin via the back-end.  If this is set to true ,the plugin can only be configured via wp-config.

## Why did it change?

Implemented the field updates as per #35 

## Did you fix any specific issues?

#36 
#33 
#26 
#13 

## CERTIFICATION

By opening this pull request, I do agree to abide by
the [CODE OF CONDUCT](https://github.com/aspirepress/.github/CODE_OF_CONDUCT.md) and be bound by the terms
of the [Contribution Guidelines](https://github.com/aspirepress/.github/CONTRIBUTING.md) in effect on the date and time
of my contribution as proven by the
revision information in GitHub. I also agree that any previous contributions shall be deemed subject to the terms of the
version in effect on the date and time of this pull request, or any future revisions for pull requests I may submit.
Further, I certify that this work is my own, is original, does not violate the intellectual property of any other person
or entity, and I am not violating any license agreements or contracts I have with any person or entity. Finally, I agree
that this code may be licensed under any license deemed appropraite by AspirePress, including but not
limited to open source, closed source, proprietary or custom licenses, and that such license terms neither violate my
rights or my copyright to this code.

### CODE REVIEW

**Comment by asirota** [comment]
I’ll have a look later today.

**Review by smileBeda — COMMENTED** [review]
In general going down the road, you will want more "detailed" @package tags in comments such as:
- `* @package           Plugins\PluginName` (where `Plugins` is used in ALL our plugins, `PluginName` is dedicated to the plugin
- `* @package           Themes\ThemeName` (same as above)

This is because if you ever plan to use PHPDOC, it will be immensely helpful to "bundle" docs in the sidebar menus of PHP DOC that get generated (then you have a "parent" menu called "Plugins" and submenus with each plugin name, in example).
Ideally that same pattern is taken down to namespaces too.

Similarly function/method class comments should follow the same @package rule (and not use @subpackage), have @author, @since (if which there can be more than one), @param (of which special case is given to Filters and Actions, again please let me know in slack if you need the syntax for those to comply with PHP DOC. I can also share VS Code setups that do automate some of this tedious stuff.)

Further it is very helpful to have declarations in method signatures like `function name( string $string = '' ): string`. PHPDOC picks these up, and it can be very helpful also for code safety in general.

This are just recommendations, I understand at early stages ... etc :)
yet the earlier we have this inside the code, the better for us later.

File: aspire-update.php
[the remark this answers is not in the document]
Comment (namithj) [reply aspire-update.php:1]: The plugin entry file name was changed as per WPCS.  It needs to be this one or it will throw an PHPCS error.

**Comment by namithj** [comment]
@smileBeda We will just be having this one Wordpress Plugin.

File: aspire-update.php
[the remark this answers is not in the document]
Comment (smileBeda) [reply aspire-update.php:1]: Yes. I am referring to the plugin file. Usually the plugin name and plugin file are the same.
`Aspire Press`
`aspire-press.php`
Or
`AspirePress`
`aspirepress.php`

**Review by smileBeda — APPROVED** [review]

File: aspire-update.php
[the remark this answers is not in the document]
Comment (namithj) [reply aspire-update.php:1]: It needs to be aspire-press.php

**Comment by asirota** [comment]
AspireUpdate is now the official name of the plugin for WordPress sites that want to use an alternative AP repository.

**Pull request merged**